cunumeric/deferred.py

```diff
@@ -171,35 +206,41 @@ class DeferredArray(NumPyThunk):
     :meta private:
     """
 
-    def __init__(self, runtime, base, dtype, numpy_array=None) -> None:
+    def __init__(
+        self,
+        runtime: Runtime,
+        base: Any,
```

Comment:
This is definitely of type `Store`, but setting that causes a bunch of warnings later.
(Leaving this here as a note for future work)